.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,88 @@
+name: Build and Publish Marketplace
+on:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    container: cimg/go:1.19
+    defaults:
+      run:
+        shell: bash
+    steps:
+    - name: build/checkout-repo
+      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      with:
+        fetch-depth: 0
+    - name: build/ensure-master-exists
+      run: git rev-parse --verify master >/dev/null 2>&1 || git branch master origin/master
+    - name: build/check-style
+      run: make check-style
+    - name: build/test
+      run: make test
+    - name: build/build
+      if: ${{ github.repository == 'mattermost/mattermost-marketplace' && (github.ref_name != 'private-beta') }}
+      run: make build
+    - name: build/build-private-beta
+      if: ${{ github.repository == 'mattermost/mattermost-marketplace' && (github.ref_name == 'private-beta') }}
+      run: BUILD_UPSTREAM_URL=https://api.integrations.mattermost.com make build
```

Comment:
It took me a second to realize what's happening between these two steps. Both steps run `make build`, but with slightly different contexts. The first runs for any branch that has these changes, besides the `private-beta` branch itself, and the second runs just for the `private-beta` branch, but with an added env var for the upstream endpoint. What's the main benefit of /not/ having the `BUILD_UPSTREAM_URL` set for the first step?

Also the second step may be a bit easier to read if we put this `BUILD_UPSTREAM_URL` in a `env` block of this step, rather than putting it in the `run` statement before `make build`. The `make build` is a bit de-emphasized at the moment.

Reply:
For the first build, it's imperative that `BUILD_UPSTREAM_URL` isn't set, since we don't want the production server making a request back to itself (or any other marketplace instance). In principle, we could make the bit to check an upstream instance not require a custom build and instead do this at run time -- back in the day, this was easiest to me :p

Your second suggestion is much appreciated! I've updated the CI to use `env: ` to fix this -- thank you :)